lib/core/tools/text_tool/text_tool.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:paintroid/core/commands/graphic_factory/graphic_factory.dart';
+import 'package:paintroid/core/tools/text_tool/bounding_box.dart';
+import 'package:paintroid/core/tools/tool.dart';
+
+class TextTool extends Tool {
+  bool isRotating;
+  BoundingBox boundingBox;
+
+  TextTool({
+    required super.commandManager,
+    required super.commandFactory,
+    required this.graphicFactory,
+    required super.type,
+    required this.boundingBox,
+    super.hasAddFunctionality = false,
+    super.hasFinalizeFunctionality = true,
+    this.isRotating = false,
+  });
+
+  String? currentText;
```

Comment:
is there a reason it being nullable? Can it no be initialized to a default text or simply just ''